frontend/src/lib/k8s/cluster.ts

```diff
@@ -157,18 +165,56 @@ export function makeKubeObject<T extends KubeObjectInterface | KubeEvent>(
 
     static useApiList<U extends KubeObject>(
       onList: (...arg: any[]) => any,
-      onError?: (err: ApiError) => void
+      onError?: (err: ApiError) => void,
+      opts?: ApiListOptions
     ) {
+      const [objs, setObjs] = React.useState<{ [key: string]: U[] }>({});
       const listCallback = onList as (arg: U[]) => void;
-      useConnectApi(this.apiList(listCallback, onError));
+
+      function onObjs(namespace: string, objList: U[]) {
+        if (objList.length > 0) {
+          let newObjs: typeof objs = {};
+          // Set the objects so we have them for the next API response...
+          setObjs(objs => {
+            newObjs = { ...objs, [namespace || '']: objList };
+            return newObjs;
+          });
+
+          let allObjs: U[] = [];
+          Object.values(newObjs).map(currentObjs => {
+            allObjs = allObjs.concat(currentObjs);
+          });
+
+          listCallback(allObjs);
+        }
+      }
+
+      const listCalls = [];
+      if (!!opts?.namespace) {
+        let namespaces: string[] = [];
+        if (typeof opts.namespace === 'string') {
+          namespaces = [opts.namespace];
+        } else {
```

Comment:
This could help a user of this API know there is an issue with their input...
```ts
    } else if (Array.isArray(opts.namespace)) {
        namespaces = opts.namespace as string[];
    } else {
        throw Error("namespace should be a string or array of strings")
    }
```

Reply:
I will add this.